### PR TITLE
Merging to Main Branch

### PR DESCRIPTION
**Pull request by omerayyildiz**
## What type of PR is this? (check all applicable)

- [x] Added new link to category
- [x] New category added
- [ ] Fixed broken link
- [ ] Typo fix
- [ ] Readme update
- [ ] Optimization
- [ ] Documentation update
- [ ] Frontend update

## Description

- Add new sources to GD, OS(OSX), PL(C), PL(C++), and PL(Ruby).
- Merging to `main` branch with other PRs.

## Related Issues & Documents

* #74 #78 #81

### CODE REVIEW

**Comment by omerayyildiz** [comment]
Sadece belirli commitleri alamıyor muyduk ya?

**Comment by damla** [comment]
Forklayıp PR açmadığın için bütün commitler üst üste biniyor @omerayyildiz

File: docs/development-tools/vim.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+id: vim
+title: Vim
+sidebar_label: Vim
+---
+
+## Beginner
+* [Cheat Sheet](https://i.imgur.com/YLInLlY.png "Cheat Sheet")
```
Comment (damla) [inline docs/development-tools/vim.md:7]:
```suggestion
- [Cheat Sheet](https://i.imgur.com/YLInLlY.png "Cheat Sheet")
```

File: docs/development-tools/vim.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+* [The Ultimate Vim configuration](https://github.com/amix/vimrc "The Ultimate Vim configuration")
```
Comment (damla) [inline docs/development-tools/vim.md:8]:
```suggestion
- [The Ultimate Vim configuration](https://github.com/amix/vimrc "The Ultimate Vim configuration")
```

File: docs/development-tools/vim.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+id: vim
+title: Vim
+sidebar_label: Vim
+---
+
+## Beginner
```
Comment (damla) [inline docs/development-tools/vim.md:6]:
```suggestion
## Beginner

```

File: docs/development-tools/vim.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+* [Cheat Sheet](https://i.imgur.com/YLInLlY.png "Cheat Sheet")
+* [The Ultimate Vim configuration](https://github.com/amix/vimrc "The Ultimate Vim configuration")
+
+## Intermediate
+* [Vim Sence](https://github.com/hugolgst/vimsence "Vim Sence")
```
Comment (damla) [inline docs/development-tools/vim.md:11]:
```suggestion
## Intermediate

- [Vim Sence](https://github.com/hugolgst/vimsence "Vim Sence")
```

File: docs/embedded-systems/arduino.md [hunk 5]
```diff
@@ -5,6 +5,12 @@ sidebar_label: Arduino
 ---
 
 ## Beginner
+- [Arduino getting started](https://www.arduino.cc/en/Guide "Arduino getting started")
```
Comment (damla) [inline docs/embedded-systems/arduino.md:8]:
```suggestion
## Beginner

- [Arduino getting started](https://www.arduino.cc/en/Guide "Arduino getting started")
```

File: docs/graphic-design/photo-and-video.md [hunk 6]
```diff
@@ -9,3 +9,4 @@ sidebar_label: Photo and Video
 ## Intermediate
 
 ## Advanced
+- [Adobe Photoshop Scripting](https://www.adobe.com/devnet/photoshop/scripting.html)
```
Comment (damla) [inline docs/graphic-design/photo-and-video.md:12]:
```suggestion
## Advanced

- [Adobe Photoshop Scripting](https://www.adobe.com/devnet/photoshop/scripting.html)
```

File: docs/graphic-design/ui-ux-design.md [hunk 7]
```diff
@@ -5,7 +5,11 @@ sidebar_label: UI-UX Design
...
 
 ## Beginner
+- [UI/UX Nedir?](https://www.mediaclick.com.tr/tr/blog/ui-ve-ux-nedir)
 
 ## Intermediate
+- [Apple Design Resources](https://developer.apple.com/design/resources/)
+- [Material Design](https://material.io)
 
 ## Advanced
+- [Figma Avatar Creator](https://www.figma.com/file/AZ9FvrUbt5ULKSy3ckO7Db/Avatar-Illustration-System-Community?node-id=65%3A1499)
```
Comment (damla) [inline docs/graphic-design/ui-ux-design.md:15]:
```suggestion
## Beginner

- [UI/UX Nedir?](https://www.mediaclick.com.tr/tr/blog/ui-ve-ux-nedir)

## Intermediate

- [Apple Design Resources](https://developer.apple.com/design/resources/)
- [Material Design](https://material.io)

## Advanced

- [Figma Avatar Creator](https://www.figma.com/file/AZ9FvrUbt5ULKSy3ckO7Db/Avatar-Illustration-System-Community?node-id=65%3A1499)
```

File: docs/operating-systems/macos.md [hunk 8]
```diff
@@ -9,3 +9,7 @@ sidebar_label: MacOS
 ## Intermediate
 
 ## Advanced
+
+## Sources and Apps
+- [Mac Icons for Big Sur](https://macosicons.com/)
```
Comment (damla) [inline docs/operating-systems/macos.md:14]:
```suggestion
## Sources and Apps

- [Mac Icons for Big Sur](https://macosicons.com/)
```

File: docs/programming-languages/c.md [hunk 9]
```diff
@@ -5,7 +5,9 @@ sidebar_label: C
 ---
 
 ## Beginner
+- [C ve Sistem Programcıları Derneği C Notları - Necati Ergin](https://www.dropbox.com/sh/soe5wl982r9fl6w/AACR0uH1Vk_JcloBXl_yRTeqa?dl=0)
 
 ## Intermediate
 
 ## Advanced
+- [Necati Ergin Medium](https://necatiergin2019.medium.com)
```
Comment (damla) [inline docs/programming-languages/c.md:13]:
```suggestion
## Beginner

- [C ve Sistem Programcıları Derneği C Notları - Necati Ergin](https://www.dropbox.com/sh/soe5wl982r9fl6w/AACR0uH1Vk_JcloBXl_yRTeqa?dl=0)

## Intermediate

## Advanced

- [Necati Ergin Medium](https://necatiergin2019.medium.com)
```

File: docs/programming-languages/cplusplus.md [hunk 10]
```diff
@@ -7,5 +7,7 @@ sidebar_label: C++
 ## Beginner
 
 ## Intermediate
+- [Necati Ergin C++ Kaynak 2](http://elektronnot.blogspot.com/2015/11/cpp-necati-ergin.html)
```
Comment (damla) [inline docs/programming-languages/cplusplus.md:10]:
```suggestion
## Intermediate

- [Necati Ergin C++ Kaynak 2](http://elektronnot.blogspot.com/2015/11/cpp-necati-ergin.html)
```

File: docs/programming-languages/ruby.md [hunk 11]
```diff
@@ -5,7 +5,14 @@ sidebar_label: Ruby
...
 
 ## Beginner
+- [1 Videoda Ruby](https://youtu.be/p4Kq30td9VA)
+- [Ruby for Beginners](http://ruby-for-beginners.rubymonstas.org)
 
 ## Intermediate
+- [Shoesrb for GUI](https://shoesrb.com)
 
 ## Advanced
+- [Ruby with .Net](http://ironruby.net)
```
Comment (damla) [inline docs/programming-languages/ruby.md:15]:
```suggestion
## Beginner

- [1 Videoda Ruby](https://youtu.be/p4Kq30td9VA)
- [Ruby for Beginners](http://ruby-for-beginners.rubymonstas.org)

## Intermediate

- [Shoesrb for GUI](https://shoesrb.com)

## Advanced

- [Ruby with .Net](http://ironruby.net)
```

**Review by damla — APPROVED** [review]
Problem solved 👍🏼

**Pull request merged**